color/src/colorspace.rs

```diff
@@ -283,6 +283,78 @@ impl ColorSpace for DisplayP3 {
     }
 }
 
+/// 🌌 The Adobe RGB (1998) color space.
+///
+/// Adobe RGB is similar to [sRGB](`Srgb`) but has higher green chromaticity, thereby extending its
+/// gamut over sRGB on that component. It was developed to encompass typical color print gamuts.
+///
+/// Its components are `[r, g, b]` (red, green, and blue channels respectively), with `[0, 0, 0]`
+/// pure black and `[1, 1, 1]` white. The natural bounds of the channels are `[0, 1]`.
+///
+/// This corresponds to the color space in [CSS Color Module Level 4 § 10.5][css-sec] and is
+/// [characterized by the ICC][icc]. Adobe RGB is described [here][adobe] by Adobe.
+///
+/// [css-sec]: https://www.w3.org/TR/css-color-4/#predefined-a98-rgb
+/// [icc]: https://www.color.org/chardata/rgb/adobergb.xalter
+/// [adobe]: https://www.adobe.com/digitalimag/adobergb.html
+#[derive(Clone, Copy, Debug)]
+pub struct A98Rgb;
+
+impl ColorSpace for A98Rgb {
+    const TAG: Option<ColorSpaceTag> = Some(ColorSpaceTag::A98Rgb);
+
+    fn to_linear_srgb([r, g, b]: [f32; 3]) -> [f32; 3] {
+        // XYZ_to_lin_sRGB * lin_A98_to_XYZ
+        #[expect(
+            clippy::cast_possible_truncation,
+            reason = "exact rational, truncate at compile-time"
+        )]
+        const LINEAR_A98RGB_TO_SRGB: [[f32; 3]; 3] = [
+            [
+                (66_942_405. / 47_872_228.) as f32,
+                (-19_070_177. / 47_872_228.) as f32,
+                0.,
+            ],
+            [0., 1., 0.],
+            [
+                0.,
+                (-11_512_411. / 268_173_353.) as f32,
+                (279_685_764. / 268_173_353.) as f32,
+            ],
+        ];
+        matmul(
+            &LINEAR_A98RGB_TO_SRGB,
+            [r, g, b].map(|x| x.abs().powf(563. / 256.).copysign(x)),
```

Comment:
Should this be `const { (563. / 256.) as f32 }`, similarly to the values above. That's a lot noisier than the current form...

Reply:
The above form requires the explicit cast to f32 so the compiler does the calculation (inside the parentheses) in f64, truncating to f32 at the last step. The reason for that is that the numerator and denominator have more significant digits than fit in a f32, so doing the entire thing in f32 may cause a very small, but avoidable, error.

For example, 

```
(-19_070_177. / 47_872_228.) as f32 -> #becbf549
-19_070_177f32 / 47_872_228f32      -> #becbf548
```

Reply:
Right, makes sense. I was just pattern matching from Raph's approval comment).